lib/Conversion/TorchToTcp/TcpCustomOp.cpp

```diff
@@ -117,6 +117,147 @@ class ConvertAtenConvolutionOp : public OpConversionPattern<AtenConvolutionOp> {
   }
 };
 
+class ConvertAtenFakeQuantizePerTensorAffineOp
+    : public OpConversionPattern<AtenFakeQuantizePerTensorAffineOp> {
+public:
+  using OpConversionPattern::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(AtenFakeQuantizePerTensorAffineOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    torch_to_tcp::TorchToTcpCustomOpConversionHelper helper{op, rewriter,
+                                                            getTypeConverter()};
+    helper.addOperand("self", adaptor.getSelf());
+    helper.addFloatAttr("scale", op.getScale());
+    helper.addIntAttr("zero_point", op.getZeroPoint());
+    helper.addIntAttr("quant_min", op.getQuantMin());
+    helper.addIntAttr("quant_max", op.getQuantMax());
+
+    return helper.replace();
+  }
+};
+
+class ConvertAtenFakeQuantizePerTensorAffineTensorQparamsOp
+    : public OpConversionPattern<
+          AtenFakeQuantizePerTensorAffineTensorQparamsOp> {
+public:
+  using OpConversionPattern::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(AtenFakeQuantizePerTensorAffineTensorQparamsOp op,
+                  OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    torch_to_tcp::TorchToTcpCustomOpConversionHelper helper{op, rewriter,
+                                                            getTypeConverter()};
+    helper.addOperand("self", adaptor.getSelf());
+    helper.addIntAttr("quant_min", op.getQuantMin());
+    helper.addIntAttr("quant_max", op.getQuantMax());
+
+    // scale
+    auto scaleOp = op.getScale().getDefiningOp();
+    if (!scaleOp)
+      return rewriter.notifyMatchFailure(op, "Missing scale operation");
+    auto scaleTensor = dyn_cast<torch::Torch::ValueTensorLiteralOp>(scaleOp);
+    if (!scaleTensor)
+      return rewriter.notifyMatchFailure(
+          op, "Scale operation is not ValueTensorLiteralOp");
+    auto scaleElements =
+        dyn_cast<DenseFPElementsAttr>(scaleTensor.getValueAttr());
+    if (!scaleElements || scaleElements.getNumElements() != 1)
+      return rewriter.notifyMatchFailure(op, "Unsupported scale type or size");
+    auto scale = (*scaleElements.begin()).convertToDouble();
+    helper.addDenseFloatArrayAttr("scale", {scale});
```

Comment:
Just a thought. Scale and zero-point are tensors in the Torch op definition. That means they might be dynamic values. By converting them to attributes, are we making the assumption here that we only support the case when they are static? Or do we want to preserve them as tensors?
```
    AnyTorchTensorType:$scale,
    AnyTorchTensorType:$zero_point,
```
Looping @navahgar on this as I recall him having some opinions on this.

Reply:
there are 2 modes for quantization op: per_tensor and per_channel.

For the per_tensor mode, a single value of scale and zero_point are used to represent the entire tensor. 

- For `AtenFakeQuantizePerTensorAffineOp` , these two values are constant float and int. 
- For the `AtenFakeQuantizePerTensorAffineTensorQparamsOp`, both scale and zero_point operands need to be size-1 ValueTensorLiteral (also constant but in tensor format with shape of [1])

For the per_channel mode, the number of scale and zero_point need to be the channel dimension C. I don't think these can be dynamic since the scale and zero_point need to be compile-time constant.

- For `AtenFakeQuantizePerChannelAffineOp` , both scale and zero_point operands need to be ValueTensorLiteral   constants with shape of [C]

Reply:
> For the per_channel mode, the number of scale and zero_point need to be the channel dimension C. I don't think these can be dynamic since the scale and zero_point need to be compile-time constant.

Per https://pytorch.org/docs/stable/generated/torch.fake_quantize_per_channel_affine.html it appears to allow any tensors for scale and zero-point, so they could be dynamic valued (shape is fixed by channel/feature length).

Reply:
I think @navahgar had some thoughts on this, just don't recall off the top of my head. Since he's on vacation, I'm fine with landing this with documentation, and circling back if this needs to be changed later.